exception-handling/raising-and-handling-exceptions.md

```diff
@@ -0,0 +1,306 @@
+| Exception | An event or error that may cause your program to stop if not properly handled. | error    | "The user provided the wrong data type causing the Traceback to throw a TypeError exception." |
+
+## Exceptions Are Python's Cry For Help
+
+An exception is a type of _thing_ (we can say "data type") that holds information about an error that the program encounters. Exceptions hold data like specifically what error occured and which line of code caused it.
+
+Python creates an exception whenever a syntax or runtime error occurs in a program. Python will then immediately try to stop the execution of the program and exit out of it. When an error happens, and we see details in the stack trace, we actually see details about the exception!
+
+The act of creating an exception is often called **raising an exception.**
+
+When an exception is raised, Python moves "up" through the stack of code execution. This act of the exception moving "up" is often called **throwing an exception** or **bubbling up.** We might imagine an exception raised in a deep layer of code gets thrown up through the each layer of code, eventually to the programmer's face.
+
+Python raises exceptions because exceptions are Python's cry for help. Python only raises exceptions when some code says, "I can't do this! I can't go on!"
+
+Of course, when Python surrenders and throws an exception, we can't force it to keep going. Ideally, however, should we see an exception, we'd love to recognize it, handle the situation, and then allow our program to move on.
+
+Code that recognizes a thrown exception is often called **catching an exception.** Catching an exception and then handling the situation, setting things back in order, and making it possible to move on, is often called **handling an exception** or **rescuing an exception.**
+
+It is wise and ideal to write code that rescues and helps out our program, should Python ever raise an exception.
+
+## Common Exception Types
+
+| Exception Name    | When it occurs                                                                      |
+| ----------------- | ----------------------------------------------------------------------------------- |
+| NameError         | Raised when a local or global name is not found.                                    |
+| ZeroDivisionError | Raised when the second argument of a division or modulo operation is zero.          |
+| OverflowError     | Raised when the result of an arithmetic operation is too large to be represented.   |
+| SyntaxError       | Raised when the parser encounters a syntax error.                                   |
+| TypeError         | Raised when an operation or function is applied to an object of inappropriate type. |
+| KeyboardInterrupt | Raised when the user hits the interrupt key (normally Control-C or Delete).         |
```

Comment:
I'll add/adjust this manually; I think since the point of the passage is to communicate "these exceptions exist" (and handling has barely been talked about), I'll go for simpler examples

Reply:
Yeah, there's definitely a risk of front loading so much in an overview that the reader shuts down.